modules/product/src/main/java/com/opengamma/strata/product/SecurityTrade.java

```diff
@@ -5,97 +5,67 @@
  */
 package com.opengamma.strata.product;
 
-import com.opengamma.strata.basics.Trade;
-import com.opengamma.strata.collect.id.LinkResolutionException;
-import com.opengamma.strata.collect.id.LinkResolver;
-import com.opengamma.strata.collect.id.LinkResolvable;
+import com.opengamma.strata.basics.currency.Currency;
 
 /**
- * A trade that is directly based on a security.
+ * A trade representing the purchase or sale of a security,
+ * where the security is embedded ready for mark-to-market pricing.
  * <p>
- * A security trade is a {@link Trade} that contains a reference to a {@link Security}.
- * The security is held via a {@link SecurityLink}, which allows for the security to be
- * managed as reference data, separately from the trade.
+ * This trade represents a trade in a security, defined by a quantity and price.
+ * The security is embedded directly, however the underlying product model is not available.
+ * The security may be of any kind, including equities, bonds and exchange traded derivative (ETD).
  * <p>
  * Implementations of this interface must be immutable beans.
- * 
- * @param <P>  the type of the product
  */
-public interface SecurityTrade<P extends Product>
-    extends FinanceTrade, LinkResolvable<SecurityTrade<P>> {
+public interface SecurityTrade
```

Comment:
This feels like another oddity. My understanding of the model was that there was a clean separation between securities (linked by ID) and products (directly referenced). This has a security which is directly referenced which confuses the picture slightly. 

What's it for? Would it be easier to explain the trade / security / product model without this?